File: applications/EmployeeNetCoreApp/EmployeeNetCoreApp/Exceptions/EntityOutdatedException.cs
using System;
using System.Data.Common;

namespace EmployeeNetCoreApp.Exceptions
{
    [Serializable]
    public class EntityOutdatedException: Exception
	{
		public string UUID { get; }
        public string? UpdatedBy { get; }
        public DateTime? UpdatedDate { get; }
        public int LocalVersion { get; }
        public int LastVersion { get; }

        public EntityOutdatedException(string UUID, string? UpdatedBy, DateTime? UpdatedDate, int LocalVersion, int LastVersion)
		{
			this.UUID = UUID;
			this.UpdatedBy = UpdatedBy;
			this.UpdatedDate = UpdatedDate;
			this.LocalVersion = LocalVersion;
			this.LastVersion = LastVersion;
		}

        public new string Message()
        {
            return string.Format("The local version of employee {0} is out of date. The most up-to-date version is {1}, updated by {2} on {3} and local version is {4}. Please update your local version", UUID, LastVersion, UpdatedBy, UpdatedDate, LocalVersion) ;
        }
    }
}

File: applications/EmployeeQuarkusApp/src/main/java/com/redhat/developers/exception/EntityOutdatedException.java
package com.redhat.developers.exception;

import java.text.SimpleDateFormat;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.Objects;

import lombok.Getter;

@Getter
public class EntityOutdatedException extends RuntimeException{

    private final String uuid;
    private final  String  updatedBy ;
    private final LocalDateTime updatedDate;
    private final Integer localVersion;
    private final Integer lastVersion;

    public EntityOutdatedException(String uuid, String updatedBy, LocalDateTime updatedDate, Integer localVersion, Integer lastVersion) {
        this.uuid = uuid;
        this.updatedBy = updatedBy;
        this.updatedDate = updatedDate;
        this.localVersion = localVersion;
        this.lastVersion = lastVersion;
    }

    @Override
    public String getMessage(){
        return String.format("The local version of employee %s is out of date. The most up-to-date version is %d, updated by %s on %s and local version is %d. Please update your local version", uuid, lastVersion, Objects.requireNonNullElse(updatedBy,"TestUser"),Objects.requireNonNullElse(updatedDate,  LocalDateTime.now()) , localVersion);
    }
}
